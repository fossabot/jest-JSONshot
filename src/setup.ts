import {toMatchSnapshot} from "jest-snapshot";
import {jsonShot} from "./lib";
import serializer from "./serialize";

declare global {
    namespace jest {
        interface Matchers<R> {
            toMatchJSONshot(): R;
        }
    }
}

export const extension: jest.ExpectExtendMap = {
    toMatchJSONshot(received)
    {
        let snap = jsonShot(received);
        return toMatchSnapshot.call(this, snap);
    }
};

expect.addSnapshotSerializer(serializer);
expect.extend(extension);