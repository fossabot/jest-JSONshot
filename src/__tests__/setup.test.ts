import "../setup";


describe('serializer', () =>
{

    test('JSONshot object', () =>
    {
        let obj = {
            key1: "val1"
        };

        expect(obj).toMatchJSONshot();
    });

    test('JSONshot string', () =>
    {
        let obj = "str1";

        expect(obj).toMatchJSONshot();
    });

    test('JSONshot number', () =>
    {
        let num = 1;

        expect(num).toMatchJSONshot();
    });

    test('regular snapshot still works', () =>
    {
        let obj = {
            key1: "val1"
        };

        expect(obj).toMatchSnapshot();
    });

    test('function', () =>
    {

        let func = function()
        {
            return "result1"
        };

        let result = expect(func).toMatchJSONshot;

        expect(result).toThrow();
    });
});