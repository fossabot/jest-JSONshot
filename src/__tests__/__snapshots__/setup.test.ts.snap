// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer JSONshot number 1`] = `1`;

exports[`serializer JSONshot object 1`] = `
{
  "key1": "val1"
}
`;

exports[`serializer JSONshot string 1`] = `"str1"`;

exports[`serializer function 1`] = `null`;

exports[`serializer regular snapshot still works 1`] = `
Object {
  "key1": "val1",
}
`;
